pkg/option/config.go

```diff
@@ -3279,6 +3283,18 @@ func (c *DaemonConfig) Populate(vp *viper.Viper) {
 			ToFQDNsMaxDeferredConnectionDeletes)
 	}
 	switch {
+	case vp.IsSet(ToFQDNsActiveConnectionsTTL): // set by user
+		if vp.GetInt(ToFQDNsActiveConnectionsTTL) < int(2*defaults.DNSGCJobInterval) {
+			c.ToFQDNsActiveConnectionsTTL = vp.GetInt(ToFQDNsActiveConnectionsTTL)
+		} else {
+			log.Infof("Using the default value for %s: %d as user value %d is smaller than the default one", ToFQDNsActiveConnectionsTTL, defaults.ToFQDNsActiveConnectionsTTL, defaults.ToFQDNsActiveConnectionsTTL)
```

Comment:
Couple of things here:
- The third parameter is described in the string as a "user value" but the parameter is a hardcoded default
- Standard style in Cilium is to prefer structured logging and avoid formatting variations on logging functions like `Infof`, `Warningf`, `Errorf`. We may make exceptions to this in the case of hardcoded values (most commonly under `daemon/cmd/`), but when some fields of the log may be variable, they should be structured:

```suggestion
			log.WithFields(logrus.Fields{
				logfields.Key: ToFQDNsActiveConnectionsTTL,
				logfields.Value: defaults.ToFQDNsActiveConnectionsTTL,
				logfields.InvalidValue: vp.GetInt(ToFQDNsActiveConnectionsTTL),
			}).Info("Invalid configuration value for key, using default value")
```

Reply:
^ We try to standardize the way that log fields are exposed to the user by defining them in the `logfields` package. `InvalidValue` isn't a logfield in that package yet, but I think the above would be a reasonable addition for this sort of case. Previously I think we may have just open-coded the string like `"invalidValue"` or `"oldValue"` but the whole idea with the `logfields` package is to ensure that logging fields are consistent over time.